### PR TITLE
use travis or circle build number

### PR DESCRIPTION
**Pull request by serge17**
Now the plugins uses travis or circle build number as a versionQualifier if available.

### CODE REVIEW

File: src/main/groovy/com/contrastsecurity/InstallContrastAgent.groovy [hunk 1]
```diff
@@ -50,6 +68,27 @@ class InstallContrastAgent extends DefaultTask {
             if (!argLineContainsContrastArgLine(property)) {
                 config.setProperty(jvmArgs, property + " " + ContrastGradlePlugin.buildArgLine(project))
                 layout.save(new FileWriter(gradleProperties, false))
+            } else {
```
Comment (gmoore) [inline src/main/groovy/com/contrastsecurity/InstallContrastAgent.groovy]: I admire what you're trying to do here but it's getting a little hard to follow. For the sake of saving us from all this complexity, let's just do this:

if `argLineContainsContrastArgLine` is true, then just do nothing. This means the user specified their own contrast argLine. We're not going to mess with it.

**Review by gmoore — CHANGES_REQUESTED** [review]
Need tests.

File: src/main/groovy/com/contrastsecurity/InstallContrastAgent.groovy (hunk 1 above)
Comment (serge17) [reply src/main/groovy/com/contrastsecurity/InstallContrastAgent.groovy]: @gmoore Currently the appVersionQualifier is generated each time the install task is run, so it will be different from the one in argline if we don't update it each time. Should we take it from the argline instead if its present?

**Comment by serge17** [comment]
@gmoore Tests are added

File: src/main/groovy/com/contrastsecurity/InstallContrastAgent.groovy (hunk 1 above)
Comment (gmoore) [reply src/main/groovy/com/contrastsecurity/InstallContrastAgent.groovy]: I don't understand that the argLine persists between runs but the appVersionQualifier does not.

File: src/test/groovy/com/contrastsecurity/ContrastPluginTest.groovy [hunk 2]
```diff
@@ -65,4 +67,24 @@ class ContrastPluginTest {
         assertEquals(argLine, actualArgLine)
     }
 
+    @Test
+    public void testComputeAppVersionQualifier() {
+
+        String travisBuildNumber = System.getenv("TRAVIS_BUILD_NUMBER")
```
Comment (gmoore) [inline src/test/groovy/com/contrastsecurity/ContrastPluginTest.groovy]: Let's rework this test.

Unit tests shouldn't depend on environment variables being set. Set the environment in your test

Create a test where TRAVIS_BUILD_NUMBER (or the corresponding variable in the plugin) is set. Call `computeAppVersionQualifier` and test that the output of `computeAppVersionQualifier` is what we would expect.

Do the same for:
* CIRCLE_BUILD_NUM
* None set at all
* Both TRAVIS_BUILD_NUMBER and CIRCLE_BUILD_NUM set

**Review by gmoore — CHANGES_REQUESTED** [review]

File: src/test/groovy/com/contrastsecurity/ContrastPluginTest.groovy (hunk 2 above)
Comment (serge17) [reply src/test/groovy/com/contrastsecurity/ContrastPluginTest.groovy]: @gmoore Done, added more tests

File: src/main/groovy/com/contrastsecurity/InstallContrastAgent.groovy (hunk 1 above)
Comment (serge17) [reply src/main/groovy/com/contrastsecurity/InstallContrastAgent.groovy]: @gmoore Now, appVersionQualifier is not changed if it is already set. If argLine is set, it is not changed.

**Review by gmoore — APPROVED** [review]
